src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx

```diff
@@ -34,43 +36,75 @@ export const OrderHistoryTransaction = ({ tx }: { tx: Transaction }) => {
   const { chainId } = useActiveWeb3React()
   const theme = useTheme()
 
-  const isSwapTransaction = tx.type === TransactionTypes.Swap
+  const isSwapTransaction = tx.type === TransactionType.Swap
+  const isLimitOrderTransaction = tx.type === TransactionType.LimitOrder
 
-  const protocol = isSwapTransaction ? tx.swapProtocol ?? 'Swapr' : 'Socket'
-  const transactionType = isSwapTransaction ? tx.type : 'Bridge Swap'
+  const formattedSellToken = useToken(tx.sellToken.tokenAddress)
+  const formattedBuyToken = useToken(tx.buyToken.tokenAddress)
+
+  // Find the protocol
+  let protocol = 'Unknown'
+  if (isSwapTransaction) {
+    protocol = tx.swapProtocol ?? 'Unknown'
+  } else if (isLimitOrderTransaction) {
+    protocol = 'COW'
+  }
+
+  // Transaction type can be an AMM order or a COW order which is can also be a market, limit or stop limit order
+  let transactionType: string = tx.type
+
+  // Transaction timestamp
+  let formattedTimestamp: string | null = null
+
+  if (tx.confirmedTime) {
+    console.log('tx.confirmedTime', tx.confirmedTime)
+    formattedTimestamp = dayjs(tx.confirmedTime).format('DD/MM/YYYY HH:mm')
+  }
+
+  let sellTokenAmount = tx.sellToken?.value
+  let buyTokenAmount = tx.buyToken?.value
+
+  // If the transaction has a creation date and a signature, it's a COW order (safe check)
+  if ('creationDate' in tx && 'signature' in tx) {
+    transactionType = tx.class.toString()
+
+    // Fix the token amounts
```

Comment:
it looks like this is more formatting than fixing

```suggestion
    // format the token amounts
```

Reply:
Addressed in https://github.com/ImpeccableHQ/swapr-dapp/pull/1646/commits/83426e029f16a9fc9ce601d3cecea919e55cd64f